modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -604,4 +610,272 @@ private void deleteApplication(String applicationId) throws Exception {
+        }
+    }
+
+    @DataProvider(name = "paginationLimitsDataProvider")
+    public Object[][] paginationLimitsDataProvider() {
+
+        return new Object[][]{
+                {10},
+                {20},
+                {25},
+        };
+    }
+
+    @Test(dataProvider = "paginationLimitsDataProvider", dependsOnMethods = "createOrganizationsForPaginationTests")
+    public void testGetPaginatedOrganizationsWithLimit(int limit) {
+
+        String endpointURL = ORGANIZATION_MANAGEMENT_API_BASE_PATH + QUESTION_MARK + LIMIT_QUERY_PARAM + limit;
+        Response response = getResponseOfGetWithOAuth2(endpointURL, m2mToken);
+
+        validateHttpStatusCode(response, HttpStatus.SC_OK);
+
+        int actualOrganizationCount = response.jsonPath().getList(ORGANIZATIONS_PATH_PARAM).size();
+        int expectedOrganizationCount = Math.min(limit, NUM_OF_ORGANIZATIONS_FOR_PAGINATION_TESTS);
+        Assert.assertEquals(actualOrganizationCount, expectedOrganizationCount);
+
+        String nextLink = response.jsonPath().getString(
+                String.format("links.find { it.%s == '%s' }.%s", REL, LINK_REL_NEXT, HREF));
+
+        String afterValue = null;
+
+        if (nextLink != null && nextLink.contains(AFTER_QUERY_PARAM)) {
+            afterValue = nextLink.substring(nextLink.indexOf(AFTER_QUERY_PARAM) + 6);
+        }
+
+        String storedAfterValue = afterValue;
+
+        if (NUM_OF_ORGANIZATIONS_FOR_PAGINATION_TESTS > limit) {
+            Assert.assertNotNull(storedAfterValue);
+        } else {
+            Assert.assertNull(storedAfterValue);
+        }
+    }
+
+    @DataProvider(name = "paginationLimits")
+    public Object[][] paginationLimits() {
+
+        return new Object[][]{
+                {1}, {2}, {5}, {6}, {10}, {17}
+        };
+    }
+
+    @Test(dependsOnMethods = "createOrganizationsForPaginationTests", dataProvider = "paginationLimits")
+    public void testGetPaginatedOrganizationsWithCursor(int limit) {
+
+        String after;
+        String before;
+
+        // Step 1: Call the first page.
+        String firstPageUrl =
+                ORGANIZATION_MANAGEMENT_API_BASE_PATH + QUESTION_MARK + LIMIT_QUERY_PARAM + limit + AMPERSAND +
+                        RECURSIVE_QUERY_PARAM + FALSE;
+        Response firstPageResponse = getResponseOfGetWithOAuth2(firstPageUrl, m2mToken);
+
+        validateHttpStatusCode(firstPageResponse, HttpStatus.SC_OK);
+
+        List<Map<String, String>> firstPageLinks = firstPageResponse.jsonPath().getList(LINKS_PATH_PARAM);
+        after = getLink(firstPageLinks, LINK_REL_NEXT);
+        before = getLink(firstPageLinks, LINK_REL_PREVIOUS);
+
+        Assert.assertNotNull(after, "After value should not be null on the first page.");
+        Assert.assertNull(before, "Before value should be null on the first page.");
+
+        // Validate the first page organizations
```

Comment:
```suggestion
        // Validate the first page organizations.
```
Check other places as well.

Reply:
Done